src/main/java/com/keniu/services/UserServiceImpl.java

```diff
@@ -19,12 +20,17 @@ public class UserServiceImpl implements UserService {
     private final RoleService roleService;
     private final UserMapper userMapper;
     private final PasswordEncoder passwordEncoder;
+    private final ShoppingCartService shoppingCartService;
 
     @Override
     public UserDto save(CreateUserRequestDto createUserRequestDto) {
         User user = userMapper.toModel(createUserRequestDto);
         user.setPassword(passwordEncoder.encode(user.getPassword()));
         user.setRoles(Set.of(roleService.findByName(RoleName.ROLE_USER)));
-        return userMapper.toDto(userRepository.save(user));
+        user = userRepository.save(user);
+        ShoppingCart shoppingCart = new ShoppingCart();
+        shoppingCart.setUser(user);
+        shoppingCartService.save(shoppingCart);
+        return userMapper.toDto(user);
```

Comment:
create method in ShopCartService `createShoppingCart(user);`

```suggestion
        userRepository.save(user);
        shoppingCartService.createShoppingCart(user);
        return userMapper.toDto(user);
```

Reply:
For two simply lines
        ShoppingCart shoppingCart = new ShoppingCart();
        shoppingCart.setUser(user);
separately method in interface and his implementation?

Reply:
logic to create shop cart for user is related to ShopCartService, not to method register in UserService

btw 3 lines ))) 
```
ShoppingCart shoppingCart = new ShoppingCart();
shoppingCart.setUser(user);
repository.save(shoppingCart);
```

Reply:
Are you kidding me?

Reply:
It looks like only you are doing this